src/09_output_housing_value.R

```diff
@@ -0,0 +1,466 @@
+
+  # Return the table with columns ordered by our defined logic
+  dplyr::tbl(con, dbplyr::in_schema(schema, table)) %>%
+    dplyr::select(dplyr::all_of(cols_sorted))
+}
+
+# ?????????????????????????????????????????????????????????????????????????????
+# do we need the JURISDICTION? It causes many problem. We only need the DA.
+# ?????????????????????????????????????????????????????????????????????????????
+
+# Initialize combined dataframes before the loop
+combined_ave_p <- NULL
+combined_med_p <- NULL
+
+# start working on the property value tables
+for (year_name in names(bca_property_values_table)[1]) {
+  # Get the table name for this year
+  current_table <- bca_property_values_table[[year_name]]
+
+  cat(sprintf(
+    "Processing property values for year %s from table %s\n",
+    year_name,
+    current_table
+  ))
+
+  # Step 2: Get property details for the current year
+  # tbl(con, in_catalog(config$database$database, "Prod", current_table))
+  # bca_properties <-tbl(con, in_schema("Prod", current_table))
+
+  bca_properties <- tbl_long_cols_mssql(
+    con,
+    "Prod",
+    current_table
+  )
+  # or just simply move JURISDICTION to the end of the query
+  # bca_properties %>%
+  #   dplyr::select(
+  #     GEN_PROPERTY_SUBCLASS_DESC,
+  #     FOLIO_ID,
+  #     GEN_GROSS_IMPROVEMENT_VALUE,
+  #     GEN_GROSS_LAND_VALUE,
+  #     JURISDICTION
+  #   ) %>%
+  #   glimpse()
+
+  bca_properties <- bca_properties %>%
+    mutate(
+      GEN_GROSS_IMPROVEMENT_VALUE = as.numeric(GEN_GROSS_IMPROVEMENT_VALUE),
+      GEN_GROSS_LAND_VALUE = as.numeric(GEN_GROSS_LAND_VALUE)
+    ) %>%
+    dplyr::filter(
+      GEN_PROPERTY_SUBCLASS_DESC %in%
+        c('Residential Single Family', 'Residential Strata'),
+      GEN_GROSS_IMPROVEMENT_VALUE > 50000,
+      GEN_GROSS_LAND_VALUE > 50000
+    ) %>%
+    mutate(
+      YR = year_name, # Use the year from our loop
+      TOTAL_VALUE = GEN_GROSS_IMPROVEMENT_VALUE + GEN_GROSS_LAND_VALUE
+    )
+
+  # Join the tables to create folio data for this year
+  folio_data <- bca_properties %>%
+    left_join(bca_addrs, by = "FOLIO_ID") %>%
+    left_join(gcs_data, by = c("PC" = "POSTALCODE"))
+
+  # Step 3: Calculate average property statistics by DA and jurisdiction
+  ave_p <- folio_data %>%
+    group_by(YR, DA) %>%
+    summarize(
+      N_PROPERTY = n_distinct(FOLIO_ID),
+      AVE_IMPROVEMENT_VALUE = mean(GEN_GROSS_IMPROVEMENT_VALUE, na.rm = TRUE),
+      AVE_LAND_VALUE = mean(GEN_GROSS_LAND_VALUE, na.rm = TRUE),
+      AVE_TOTAL_VALUE = mean(TOTAL_VALUE, na.rm = TRUE),
+      MIN_TOTAL_VALUE = min(TOTAL_VALUE, na.rm = TRUE),
+      MAX_TOTAL_VALUE = max(TOTAL_VALUE, na.rm = TRUE),
+      .groups = "drop"
+    )
+
+  ave_p <- ave_p %>%
+    collect()
+
+  # ave_p %>% count(DA) %>% filter(n > 1)
+  # ave_p %>%
+  #   filter(DA == "010120219")
+  # 1063 DAs in 2023 has at least two JURISDICTIONs. It causes another problem.
+
+  # Step 4: Calculate median property values
+  med_p <- folio_data %>%
```

Comment:
Is there a reason this is done as a separate table instead of just another column in the average table? Seems like it's the same logic/could be done all at once.